docs/rfc/0002-processor-interface.md

```diff
@@ -0,0 +1,110 @@
+# Processor Interface RFC
+
+- Contribution Name: Processor Interface/Plugin Framework
+- Implementation Owner: Stephen Akiki
+
+## Problem Statement
+
+Users of the Indexer each have their own needs when it comes to the data gathered by the Indexer. Whether it is
+extracting certain fields of information or transforming it entirely, the data gathered is something that every user
+consumes in a different way.
+
+Currently, users must "drink from a firehouse" of information in that all information is presented at all times. No
+custom "pre-processing" of the data can be done so no derivative information can be created. This means that the user
+must save all the data and only after that is accomplished can they use the information. This can result in an
+incredible amount of "wasted" storage if the user only needs to save a derived value and not all underlying data.
+
+Additionally, there might be various transformations that are useful enough that one would wish to "pipe" them from one
+to another, similar to how commands are piped in Linux. This would allow for modules to be saved and re-used across
+Indexer deployments.
+
+## Proposal
+
+### Plugin Interface
+
+A sample interface and pipeline effort is shown below:
+
+```go
+package processor
+
+// Config will act as a placeholder for now. It will end up providing an interface for
+// serialization/deserialization of config files.
+// Derived types will provide plugin-specific data fields.
+type Config string
+
+type MetaData string
+
+type Processable interface {
+    // Metadata associated with each Processor.
+    Metadata() MetaData
+
+    // Config returns the configuration options used to create an Processor.
+    // Initialized during `Connect`, it will return nil until the Exporter has been Connected.
+    Config() Config
+
+    // Init will be called during initialization, before block data starts going through the pipeline.
+    // Typically used for things like initializating network connections.
+    // The Config passed to Connect will contain the Unmarhsalled config file specific to this plugin.
+    Init(cfg Config) error
```

Comment:
We probably need to pass a `context.Context` into this so that we can terminate early if needed.

Reply:
Are we going to give plugins their own data directories, or does the plugin need a config for providing the data directory?

Reply:
I've started writing some code to lookup plugin configs in the indexer data dir. However, I think we're going to want to end up just having a single Conduit config (like `conduit.yml` or something) which specifies an entire pipeline and all the necessary config values instead of requiring users to specify a different config file for each plugin they're using.

Reply:
I think Eric's approach is the most flexible.

I can add the context into the function @winder